pkg/driver/clickhouse/clickhouse.go

```diff
@@ -95,6 +99,14 @@ func (drv *Driver) openClickHouseDB() (*sql.DB, error) {
 	return sql.Open("clickhouse", clickhouseURL.String())
 }
 
+func (drv *Driver) onClusterClause() string {
+	clusterClause := ""
+	if drv.clusterParameters.OnCluster {
+		clusterClause = fmt.Sprintf(" ON CLUSTER '%s'", drv.clusterParameters.ClusterMacro)
```

Comment:
It's unfortunate that we can't use `drv.quoteIdentifier()` here, because these are strings and not identifiers, which are quoted/escaped differently.

It is probably worth introducing a new `drv.escapeString()` function that escapes `'` as `\'` and `\` as `\\`, per [the ClickHouse documentation on strings](https://clickhouse.com/docs/en/sql-reference/syntax#string), and use that here, and anywhere else we have string values that need escaping.

Reply:
I have added drv.escapeString method + a test for it.

I liked it more as escaping method then a quoting method, so I made the return value be the escaped string, while keeping the '' around the %s for it to be explicit.